### DISCUSSION TITLE
empty().separated_by(just(";")) making no progress

### DISCUSSION THREAD

**Discussion by apps4uco** [zesterer/chumsky#404] [Q&A] [answered]
I have a parser that I want to parse a list of values, that can also be empty values.
I am using the latest github 1.0 alpha

I made a simple test case that reproduces the problem
```
fn empty_separated<'a>() -> impl Parser<'a, &'a str,
        Vec<()>, extra::Err<Rich<'a, char>>> {
        empty().separated_by(just(";")).collect()
    }

#[test]
    fn test_empty_separated() {
        let input=";";
        let (result,errs) = empty_separated().parse(input).into_output_errors();
     }
```

When run
found Collect combinator making no progress 

Expected output
a vector of length 2

The actual code is more complex using a choice and mapping to an enum, however, I think the test code should work, or did I misunderstand something?

**Comment by zesterer** [comment, marked as answer]
This appears to be a bug in the no-progress detection, rather than a bug in either `empty` or `separated_by`. Compiling with `--release` has this code pass.

It seems like the no-progress detection only checks the progression of the item, not the item and separator in combination.

I've fixed this in [cfb143](https://github.com/zesterer/chumsky/commit/cfb143ad6e361225d32db7323d14f3a5b2cc48cd) and added tests for it, thanks for bringing it to my attention :)

**Reply by apps4uco** [reply]
Cool thanks
